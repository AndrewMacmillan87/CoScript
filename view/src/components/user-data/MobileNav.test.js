import React from "react";
import ReactDOM from "react-dom";
import { MemoryRouter as Router, Link } from "react-router-dom";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import { Provider } from "react-redux";
import { reduxStore } from "../../redux/store/reduxStore";

import MobileNav from "./MobileNav";

describe("MobileNav component", () => {
    test("renders without crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(
            <Provider store={reduxStore}>
                <Router>
                    <MobileNav />
                </Router>
            </Provider>,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it("renders correctly", () => {
        const tree = renderer
            .create(
                <Provider store={reduxStore}>
                    <Router>
                        <MobileNav />
                    </Router>
                </Provider>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("Should render a nav element with a className of 'profile-mobile-nav'", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <MobileNav />
                </Router>
            </Provider>
        );
        expect(wrapper.find("nav").length).toEqual(1);
        expect(wrapper.find(".profile-mobile-nav").length).toEqual(1);
    });

    it("Should render a button element with a className of 'profile-dropdown-icon-link'", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <MobileNav />
                </Router>
            </Provider>
        );
        expect(wrapper.find("button").length).toEqual(1);
        expect(wrapper.find(".profile-dropdown-icon-link").length).toEqual(1);
    });

    it("Should render 5 Link components", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <MobileNav />
                </Router>
            </Provider>
        );
        expect(wrapper.find(Link).length).toEqual(5);
    });

    it("Should render 6 a elements", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <MobileNav />
                </Router>
            </Provider>
        );
        expect(wrapper.find("a").length).toEqual(6);
    });

    it("Should render a div element with an id of 'dropdown-nav'", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <MobileNav />
                </Router>
            </Provider>
        );
        expect(wrapper.find("div").length).toEqual(1);
        expect(wrapper.find("#dropdown-nav").length).toEqual(1);
    });
});
