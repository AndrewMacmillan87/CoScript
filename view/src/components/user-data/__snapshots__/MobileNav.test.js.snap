// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileNav component renders correctly 1`] = `
<nav
  className="profile-mobile-nav"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <i
      className="home-icon fas fa-home fa-3x"
    />
  </a>
  <button
    className="profile-dropdown-icon-link"
    onClick={[Function]}
  >
    <img
      alt="Burger menu icon"
      className="profile-dropdown-icon"
      src={
        Object {
          "ReactComponent": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "__esModule": true,
          "default": "menu.svg",
        }
      }
    />
  </button>
  <div
    className="dropdown-menu dropdown-content"
    id="dropdown-nav"
  >
    <a
      href="/profile"
      onClick={[Function]}
    >
      <p
        className="dropdown-link"
      >
        My Profile
      </p>
    </a>
    <a
      href="/messages"
      onClick={[Function]}
    >
      <p
        className="dropdown-link"
      >
        Messages
      </p>
    </a>
    <a
      href="/projects"
      onClick={[Function]}
    >
      <p
        className="dropdown-link"
      >
        Active Projects
      </p>
    </a>
    <a
      href="/publications"
      onClick={[Function]}
    >
      <p
        className="dropdown-link"
      >
        Publications
      </p>
    </a>
    <a
      href=" "
      onClick={[Function]}
    >
      <p
        className="dropdown-link"
      >
        Logout
      </p>
    </a>
  </div>
</nav>
`;
